src/post/Write/EditorActions.js

```diff
@@ -19,23 +20,36 @@ export const CREATE_POST_ERROR = '@editor/CREATE_POST_ERROR';
 
 const requiredFields = 'parentAuthor,parentPermlink,author,permlink,title,body,jsonMetadata'.split(',');
 
+const getPermlink = (author, permlink) =>
+  steem.api.getContent(author, permlink).then((content) => {
+    let newPermlink = permlink;
+    if (content.id !== 0) { newPermlink = `${Date.now().toString(32)}-${permlink}`; }
+    return newPermlink;
+  }).catch((err) => {
+    console.warn('Error while getting content', err);
+    return permlink;
+  });
+
+
 export function createPost(postData) {
   requiredFields.forEach((field) => {
     assert(postData[field] != null, `Developer Error: Missing required field ${field}`);
   });
 
   return (dispatch) => {
-    const { parentAuthor, parentPermlink, author, permlink, title, body, jsonMetadata } = postData;
+    const { parentAuthor, parentPermlink, author, title, body, jsonMetadata } = postData;
     dispatch({
       type: CREATE_POST,
       payload: {
-        promise:
-        SteemConnect
-          .commentAsync(parentAuthor, parentPermlink, author, permlink, title, body, jsonMetadata)
-          .then(({ result }) => {
-            browserHistory.push(`/${parentPermlink}/@${author}/${permlink}`);
-            return result;
-          }).catch(err => err)
+        promise: getPermlink(author, postData.permlink)
+          .then(permlink =>
+            SteemConnect
+              .commentAsync(parentAuthor, parentPermlink, author, permlink, title, body, jsonMetadata)
+              .then(({ result }) => {
+                browserHistory.push(`/${parentPermlink}/@${author}/${permlink}`);
```

Comment:
It's an anti-pattern to manipulate the UI from an action. The action is supposed to changed the state and UI components should respond to state change.